src/helpers/getReactNativeHostComponentNames.tsx

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Text, TextInput, View } from 'react-native';
+import { HostComponentNames } from '../config';
+import render from '../render';
+
+export function getReactNativeHostComponentNames(): HostComponentNames {
+  const { getByTestId } = render(
+    <View>
+      <Text testID="text">Hello</Text>
+      <TextInput testID="textInput" />
+    </View>
+  );
+
+  return {
+    text: getByTestId('text').type as string,
```

Comment:
Maybe we could type the queries to return HostReactTestinstance = ReactTestInstance & { type: string} when they do return host components, it could be pretty convenient to know if they return composite or host just by looking at the type. I'll look into it a bit more

Reply:
Could we handle this using some actual type checking (`typeof x === "string"`) and not just type casting?

Reply:
Updating the queries typing doesn't look trivial at all so for now I just replaced this part with type checking and if we don't correctly detect the host component names then the user won't get a warning

Reply:
I mean we could just do something like
```
const textType = getByTestId('text').type;
if (typeof textType !== 'string') {
  throw new Error("Expected to find string type of host Text");
}
```